src/core/FileManagerServices.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * FileManagerServices.cpp - Helper file for cross platform file management
+ *
+ * Copyright (c) 2025 Andrew Wiltshire <aw1lt / at/ proton/ dot/me >
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "FileManagerServices.h"
+
+#include <QDir>
+#include <QProcess>
+
+namespace lmms {
+void FileManagerServices::openDir(QString& path)
+{
+	QString nativePath = QDir::toNativeSeparators(path);
+
+#if defined(_WIN32)
+	QProcess::startDetached("explorer", {nativePath});
+#elif defined(__APPLE__)
+	QProcess::startDetached("open", {nativePath});
+#else
+	QProcess::startDetached("xdg-open", {nativePath});
+#endif
+}
+void FileManagerServices::select(const QFileInfo item)
+{
+	QString path = QDir::toNativeSeparators(item.canonicalFilePath());
+	QStringList params;
+
+#ifdef _WIN32
+	if (!item.isDir())
```

Comment:
Hmm... I believe we established here https://github.com/LMMS/lmms/pull/7700#discussion_r1957233560 that we don't need this guard as folder selection still works great in both macOS and Windows when a directory is provided.